### DISCUSSION TITLE
Setting Staging Environments in next js

### DISCUSSION THREAD

**Discussion by kanha-jain** [vercel/vercel#5975] [Help]
Hi! In next js we have ways to set environment variables in files like .env.local, .env.production, .env.development. However, I want to set a pre-production environment or staging environment and have a separate value for variables. Kindly help me in finding a suitable way to declare those and also make some of  them available client-side

**Comment by styfle** [comment]
Hello @kanha-jain 

Take a look at this announcement and let me know if that solves your problem: https://vercel.com/changelog/environments-variables-per-git-branch
